core/src/main/java/com/google/zxing/common/StringUtils.java

```diff
@@ -29,28 +30,27 @@
  */
 public final class StringUtils {
 
-  private static final String PLATFORM_DEFAULT_ENCODING = Charset.defaultCharset().name();
-  public static final String SHIFT_JIS = "SJIS";
-  public static final String GB2312 = "GB2312";
-  private static final String EUC_JP = "EUC_JP";
-  private static final String UTF8 = "UTF8";
-  private static final String ISO88591 = "ISO8859_1";
+  private static final Charset PLATFORM_DEFAULT_ENCODING = Charset.defaultCharset();
+  public static final Charset SHIFT_JIS = Charset.forName("SJIS");
+  public static final Charset GB2312 = Charset.forName("GB2312");
+  private static final Charset EUC_JP = Charset.forName("EUC_JP");
   private static final boolean ASSUME_SHIFT_JIS =
-      SHIFT_JIS.equalsIgnoreCase(PLATFORM_DEFAULT_ENCODING) ||
-      EUC_JP.equalsIgnoreCase(PLATFORM_DEFAULT_ENCODING);
+      SHIFT_JIS.equals(PLATFORM_DEFAULT_ENCODING) ||
+      EUC_JP.equals(PLATFORM_DEFAULT_ENCODING);
 
   private StringUtils() { }
 
   /**
    * @param bytes bytes encoding a string, whose encoding should be guessed
    * @param hints decode hints if applicable
-   * @return name of guessed encoding; at the moment will only guess one of:
-   *  {@link #SHIFT_JIS}, {@link #UTF8}, {@link #ISO88591}, or the platform
-   *  default encoding if none of these can possibly be correct
+   * @return Charset of guessed encoding; at the moment will only guess one of:
+   *  {@link #SHIFT_JIS}, {@link StandardCharsets#UTF_8},
+   *  {@link StandardCharsets#ISO_8859_1}, or the platform default encoding if
+   *  none of these can possibly be correct
    */
-  public static String guessEncoding(byte[] bytes, Map<DecodeHintType,?> hints) {
+  public static Charset guessEncoding(byte[] bytes, Map<DecodeHintType,?> hints) {
```

Comment:
Likewise I think we actually have to preserve this method for compatibility, just in case. Just make a new differently-named one and have the old one call it.

Reply:
Got it. :ok_hand: